pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -218,7 +219,11 @@ def transfer_distrib_spawn_state_on_fit_end(self, results):
             # save the last weights
             last_path = None
             # TODO: is there a better way than accessing trainer through model -> trainer?
-            if not self.lightning_module.trainer.testing and best_model_path is not None and len(best_model_path) > 0:
+            if (
+                self.lightning_module.trainer.state is TrainerState.FITTING
```

Comment:
will it be a good idea to have some trainer properties here. like `is_fitting`, `is_testing` (can have better names)... just to avoid importing TrainerState everywhere

Reply:
It's not a bad idea but I would rather not do it on this PR since it is long enough already

Reply:
cool :+1: